src/js/DropContainer.jsx

```diff
@@ -0,0 +1,284 @@
+   */
+   handleChange (event) {
+      let e = event.nativeEvent,
+         dataTransferItems = this.extractData(event),
+         fileFormat = this.fileFormatFinder(this.props.accept),
+         isFileValid = this.validateFileFormat(Array.prototype.slice.call(dataTransferItems), fileFormat);
+
+      // Check to make sure the dropped file
+      // is in correct format - e.g image/*
+      if(isFileValid){
+         let eventFile = e.dataTransfer || event.target;
+
+         if(eventFile) {
+            this.dragAndDropFileUrlGenerator(eventFile)
+         } else {
+            this.inputButtonFileUrlGenerator(eventFile);
+         }
+      }
+   }
+
+   dragAndDropFileUrlGenerator (eventFile) {
+      let file = eventFile.files[0], uploadFile;
+
+      if(typeof file !== undefined) {
+         uploadFile = URL.createObjectURL(file);
+
+         this.setState({
+            fileSrc: uploadFile
+         });
+      }
+   }
+
+   inputButtonFileUrlGenerator (eventFile) {
+      let uploadFile;
+      if(typeof eventFile !== undefined) {
+         uploadFile = URL.createObjectURL(eventFile);
+
+         this.setState({
+            fileSrc: uploadFile
+         });
+      }
+   }
+
+   /**
+      * Handle file type validation
+      * make sure the file is an image
+      * @param  {Event]} event
+      * @return {object}
+   */
+   extractData (event) {
+      return event.dataTransfer && event.dataTransfer.items ?
+               event.dataTransfer.items : [];
+   }
+
+   fileFormatFinder (props) {
+      // const FILE_FORMATS = {
+      //    image: /\.jpg | \.jpeg | \.gif | \.png/,
+      //    text: /\.txt | \.doc | \.docs/,
+      //    zip: /\.zip | \.tar/,
+      //    video: /\.mov | \.mpeg | \.mpg | \.avi | \.mkv | \.wmv/,
+      //    music: /\.m1v | \.mp2 | \.mp3 | \.mpa | \.mpe | \.m3u/
+      // }
+      const FILE_FORMATS = {
+         image: 'image/*',
+         text: 'text/*',
+         pdf: 'application/pdf',
+         zip: 'application/zip',
+         video: 'video/*',
+         audio: 'audio/*',
+         app: 'application/*'
+      }
+      return FILE_FORMATS[props];
+   }
+
+   /**
+      * Handle file type validation
+      * make sure the file is an image
+      * @param  {Event]} event
+      * @return {bool}
+   */
+   validateFileFormat (file, fileFormat) {
+      return file.every(f => accepts(f, fileFormat));
+   }
+
+
+   /**
+      * Handle border color
+      * if file type is correct
+      * make border GREEN
+   */
+   correctFormatBorderStyle () {
+      this.setState({
+         dropzoneStyle: 'correctFormat'
+      });
+   }
+
+   /**
+      * Handle border color
+      * if file type is incorrect
+      * make border RED
+   */
+   incorrectFormatBorderStyle () {
+      this.setState({ dropzoneStyle: 'incorrectFormat' });
+   }
+
+   render () {
+      // define styles
+      let { dropzoneStyle, visibility } = this.state,
+         dropContainerStyle = classNames(css.dropzoneContainer, css[dropzoneStyle]);
```

Comment:
I typically follow google JS style guides and don't chain my variable definitions. Basically I make each line start with `var`, `let`, etc. I found it slightly easier to read.